skimage/segmentation/tests/test_felzenszwalb.py

```diff
@@ -61,14 +61,16 @@ def test_color():
     img = np.zeros((20, 21, 3))
     img[:10, :10, 0] = 1
     img[10:, :10, 1] = 1
+    img[:10, 10:, :] = [-1, -1, -1]  # vector with negative dot product with standard basis
     img[10:, 10:, 2] = 1
-    seg = felzenszwalb(img, sigma=0)
-    # we expect 4 segments:
-    assert_equal(len(np.unique(seg)), 4)
-    assert_array_equal(seg[:10, :10], 0)
-    assert_array_equal(seg[10:, :10], 2)
-    assert_array_equal(seg[:10, 10:], 1)
-    assert_array_equal(seg[10:, 10:], 3)
+    for similarity in ['euclidean', 'cosine']:
```

Comment:
search for `parametrize` in the test file to see how it can be used. It makes it very clear which parameter is failing when your tests fail. pytest will report exactly what (combination of) parameter caused the test to fail, and which one passed.

Reply:
This is very helpful to know. Thank you!